ignite/services/plugin/interface.go

```diff
@@ -1,44 +1,63 @@
+	gob.Register(Manifest{})
+	gob.Register(ExecutedCommand{})
+	gob.Register(ExecutedHook{})
 }
 
 // An ignite plugin must implements the Plugin interface.
+//
+//go:generate mockery --srcpkg . --name Interface --structname PluginInterface --filename interface.go --with-expecter
 type Interface interface {
-	// Commands returns one or multiple commands that will be added to the list
-	// of ignite commands. It's invoked each time ignite is executed, in
-	// order to display the list of available commands.
-	// Each commands are independent, for nested commands, use the field
-	// Command.Commands.
-	Commands() []Command
-	// Execute will be invoked by ignite when a plugin commands is executed.
-	// cmd is the executed command (one of the those returned by Commands method)
-	// args is the command line arguments passed behing the command.
-	Execute(cmd Command, args []string) error
-	// Hooks defines custom hooks registered with a given plugin
-	Hooks() []Hook
-	// ExecuteHookPre is invoked by Ignite when a command specified by the hook
-	// path is invoked is global for all hooks registered to a plugin context on
-	// the hook being invoked is given by the `hook` parameter.
-	ExecuteHookPre(hook Hook, args []string) error
-	// ExecuteHookPost is invoked by Ignite when a command specified by the hook
-	// path is invoked is global for all hooks registered to a plugin context on
-	// the hook being invoked is given by the `hook` parameter.
-	ExecuteHookPost(hook Hook, args []string) error
-	// ExecuteHookCleanUp is invoked right before the command is done executing
-	// will be called regardless of execution status of the command and hooks.
-	ExecuteHookCleanUp(hook Hook, args []string) error
+	// Manifest declares the plugin's Command(s) and Hook(s).
+	Manifest() (Manifest, error)
+
+	// Execute will be invoked by ignite when a plugin Command is executed.
+	// It is global for all commands declared in Manifest, if you have declared
+	// multiple commands, use cmd.Path to distinguish them.
+	Execute(cmd ExecutedCommand) error
+
+	// ExecuteHookPre is invoked by ignite when a command specified by the Hook
+	// path is invoked.
+	// It is global for all hooks declared in Manifest, if you have declared
+	// multiple hooks, use hook.Name to distinguish them.
+	ExecuteHookPre(hook ExecutedHook) error
+	// ExecuteHookPost is invoked by ignite when a command specified by the hook
+	// path is invoked.
+	// It is global for all hooks declared in Manifest, if you have declared
+	// multiple hooks, use hook.Name to distinguish them.
+	ExecuteHookPost(hook ExecutedHook) error
+	// ExecuteHookCleanUp is invoked by ignite when a command specified by the
+	// hook path is invoked. Unlike ExecuteHookPost, it is invoked regardless of
+	// execution status of the command and hooks.
+	// It is global for all hooks declared in Manifest, if you have declared
+	// multiple hooks, use hook.Name to distinguish them.
+	ExecuteHookCleanUp(hook ExecutedHook) error
+}
+
```

Comment:
We should document all exported types, especially in this `interface.go` file.

Reply:
yes thanks for the suggestion [651e399](https://github.com/ignite/cli/pull/3060/commits/651e3992b97dc0165ad459680ef429e195247a97)